src/classes/RD2_OpportunityEvaluationService.cls

```diff
@@ -196,26 +199,28 @@ public inherited sharing class RD2_OpportunityEvaluationService {
     /**
      * @description Extracts open installment Opportunities for the specified Recurring Donation
      * @param rd Recurring donation containing list of all related Opportunities
-     * @return List<Opportunity> Open Opportunities
+     * @param openOppsToVoid Open Opportunities to void from other Recurring Donations
+     * @return void
      */
-    private List<Opportunity> extractOpenInstallmentOpps(npe03__Recurring_Donation__c rd) {
-        List<Opportunity> opps = new List<Opportunity>();
+    private void handleClosedRecurringDonation(npe03__Recurring_Donation__c rd, List<Opportunity> openOppsToVoid) {
+
+        if (rd.Status__c != RD2_Constants.STATUS_CLOSED || isCloseActionOverride) {
```

Comment:
Even though behaviour is not directly related to this PR, wanted to verify functionality. When a user creates an open Opp for a Closed RD, it will be voided (deleted or set to Closed Lost if settings indicate so) by this service. Is this expected behaviour, ie user cannot create an open Opp for a Closed RD?

FYI @Julian88Tex